ctags/parsers/asciidoc.c

```diff
@@ -69,22 +72,25 @@ static NestingLevel *getNestingLevel(const int kind)
 static void makeAsciidocTag (const vString* const name, const int kind)
 {
 	const NestingLevel *const nl = getNestingLevel(kind);
+	int r = CORK_NIL;
 
 	if (vStringLength (name) > 0)
 	{
+		tagEntryInfo *parent = getEntryOfNestingLevel (nl);
 		tagEntryInfo e;
+
 		initTagEntry (&e, vStringValue (name), &(AsciidocKinds [kind]));
 
 		e.lineNumber--;	/* we want the line before the '---' underline chars */
 
-		if (nl && nl->type < kind)
+		if (parent && ((parent->kind - AsciidocKinds) < kind))
 		{
-			e.extensionFields.scopeKind = &(AsciidocKinds [nl->type]);
-			e.extensionFields.scopeName = vStringValue (nl->name);
+			e.extensionFields.scopeKind = &(AsciidocKinds [parent->kind - AsciidocKinds]);
```

Comment:
Indeed. Blame rst.c from uctags which I used as a template for these simple parsers. Don't blame me for not thinking while doing that :-).